src/extra_functions.jl

```diff
@@ -1,5 +1,5 @@
 @register Base.getindex(x,i::Integer) false
-@register Base.getindex(x,i) false
+@register Base.getindex(x,i) # define one and only one promotion rule
```

Comment:
what is the promotion rule?

Reply:
symtype(getindex(x, 1)) == Real

Reply:
what does the false have to do with that?

Reply:
false means not to define the promotion rule to avoid duplicate method definition.

Reply:
How does that fix it?

Reply:
Define at least on promotion rule.

Reply:
Oh, both were false. Got it.